Automattic/wp-calypso
# Hosting Configuration: Add cache notices and light refactor of API interactions

## Proposed Changes

I wanted to display notices when edge cache is enabled/disabled on the Hosting Configuration page:

![CleanShot 2024-01-02 at 07 16 12@2x](https://github.com/Automattic/wp-calypso/assets/36432/07345178-c8a0-4913-ae67-55b57434c425)

I ended up doing a bit of refactoring too:

* Cache manipulation functions and hooks now live in `client/data/hosting/use-cache.ts`.
* All of the edge cache enable/disable notifs use `EDGE_CACHE_ENABLE_DISABLE_NOTICE_ID` so they're kept in sync.
* Command palette writes to `queryClient.setQueryData( [ USE_EDGE_CACHE_QUERY_KEY, siteId ], true );` so the toggle is kept in sync.
* "Clear cache" command always calls `clearWordPressCache()`, for consistency with the Hosting Configuration card.

I wanted the commands to use `useEdgeCacheQuery` and `useSetEdgeCacheMutation` directly, but it didn't seem to technically be possible.

## Testing Instructions

1. Create a Business site and take it Atomic.
2. Navigate to the Hosting Configuration page.
3. Enable edge cache with the Command Palette. Verify the notices appear as expected, edge cache is enabled, and the state syncs with the Hosting Configuration card.
4. Disable edge cache from the card. Verify similar notices appear as expected and edge cache is disabled.
